### PR TITLE
Transform existing documentation for PORTING and USING to markdown

### PR DESCRIPTION
**Pull request by kushalsingh007**
- Converted the existing documentation to markdown format.
- Fix for #1458.

### CODE REVIEW

**Comment by kushalsingh007** [comment]
Note : Since markdown format means that I need to add to spaces at the end of a line to do line break , the static test for whitespace_check is going to fail , so I suggest you take a manual look at the commit.

**Comment by gebart** [comment]
Why do you need hard linebreaks within the paragraphs?

**Comment by kushalsingh007** [comment]
I did that to make sure the character length did not exceed 80 ( my aim was to make it look as close to the original file as possible in markup output)  while preserving the markup format. Was that a wrong assumption ? It's my first time using markup format so I might have done it wrong.

**Comment by gebart** [comment]
My recommendation is to use soft line breaks to keep the text within 80 cols when reading the text in a terminal, but let the markdown renderer reflow the text when rendering it in a rich viewer, such as GitHub.

**Comment by kushalsingh007** [comment]
Ok , then I'll change it as per your suggestion :)

**Comment by kushalsingh007** [comment]
@gebart : I think I have made the line break changes now

**Comment by kushalsingh007** [comment]
I think all changes have been made in this PR and it can be merged now. ( the fail is due to whitespace error which can't be avoided here )

**Comment by authmillenon** [comment]
@kushalsingh007 try it ;-)

```diff
diff --git a/pkg/PORTING.md b/pkg/PORTING.md
index 2acf5dd..778d77c 100644
--- a/pkg/PORTING.md
+++ b/pkg/PORTING.md
@@ -1,10 +1,10 @@
 This directory provides some porting information for libraries and programs to
 use with RIOT (to build an external module). If you'd like to add a package to
 RIOT you need to add a directory with the name of your package to this directory.
-Your directory should contain at least two files:  
-  
+Your directory should contain at least two files: 
+ 
  * patch.txt - Your patch of the upstream application of the package to make
- it build with RIOT.  
+ it build with RIOT. 
  * Makefile - A Makefile describing how to get the upstream application, apply
  the patch and how to build the package as a RIOT module.
  A rough template for several methods of acquiring a package
diff --git a/pkg/USING.md b/pkg/USING.md
index 36a0d82..1bd2cfc 100644
--- a/pkg/USING.md
+++ b/pkg/USING.md
@@ -1,5 +1,5 @@
 Packages are included to your application as external modules. Thus you only
 have to add the following line to your application (and update your INCLUDE path
-accordingly):  
+accordingly): 
 
     USEPKG += <pkg_name>
```

File: pkg/PORTING.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+This directory provides some porting information for libraries and programs to
+use with RIOT (to build an external module). If you'd like to add a package to
+RIOT you need to add a directory with the name of your package to this directory.
+Your directory should contain at least two files:  
+  
+ * patch.txt - Your patch of the upstream application of the package to make
```
Comment (authmillenon) [inline pkg/PORTING.md]: Can you do it like this?

```md
 * **patch.txt**: Your patch [...]
```

I think the readability would greatly benefit from it.

File: pkg/PORTING.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+ it build with RIOT.  
+ * Makefile - A Makefile describing how to get the upstream application, apply
```
Comment (authmillenon) [inline pkg/PORTING.md]: Here to:

```md
 * **Makefile**: A Makefile [...]
```

**Comment by kushalsingh007** [comment]
Tried your suggestion , it worked :D , thanks for clearing my misconception.

**Comment by authmillenon** [comment]
> And I think it's time for squashing now :P , because that is causing the build to fail now.

That's okay as long as your PR is under review.

Anyway: ACK, when Travis is happy. Please squash ;-).

**Comment by authmillenon** [comment]
And good night. Need to be up in 4 h ^^.

**Comment by kushalsingh007** [comment]
Done Squashing . Kindly Restart the build , the build failed because of delay in receiving input.
```
        No output has been received in the last 10 minutes, this potentially indicates a stalled build or something wrong with the build itself.

```

**Comment by gebart** [comment]
I kicked Travis for you.

**Comment by kushalsingh007** [comment]
Thanks @gebart . I think this one can be merged now :smile:

**Comment by gebart** [comment]
yes, & go!

**Pull request merged**